Autopilot/Inc/SDCard.hpp

```diff
@@ -0,0 +1,81 @@
+/**
+ * SDcard fuctions
+ * Author: Avani Joshi and Laye Mehta
+ */
+
+#ifndef SDCARDHEADER_HPP
+#define SDCARDHEADER_HPP
+
+#include "SPI.hpp"
+
+struct SDCardData_t {
+	
+	bool isRead; 
+ 
+    bool status; //Success = 1 and Fail = 0
+};
+
+
+class SDCard{
+    public:
+	
+	/*
+	THE FUNCTION ARGUMENTS WILL BE ADDED AS THE IMPLEMENTATION IS CREATED
+	*/
+		
+		/*
+		Reads from SD Card
+		*/
+		virtual void Read() = 0; 
+		
+		/*
+		Writes to SD Card
+		*/
+		virtual void Program() = 0; 
```

Comment:
Yeah, to Sahil's point, if communicating with all SD cards is the same, then we probably don't need to have the abstract class.

In case you weren't clear about it, half the reason we have abstract classes is to allow for unit testing against modules (but since your driver is an "optional" feature that doesn't live in the core autopilot architecture, I don't see why anyone would unit test against it).

The other half is that for sensors, if we were to use a different airspeed sensor for example, then we could just create a second derived class from the abstract class for that sensor. The benefit to that being that the person using the driver isn't affected by which sensor is actually used; they just happily call the same base class functions. But since it seems all sd cards are the same (though pls correct me if I'm wrong), no need.